api.php

```diff
@@ -1,12 +1,57 @@
 <?php
 require_once('Books.php');
-if (preg_match("[^/api/books(|/)$]", $_SERVER['REQUEST_URI'])) {
-    Books::getInstance()->allBooks();
+function get($value, $default = null)
+{
+    return isset($value) ? $value : $default;
 }
-if (preg_match("[^/api/books/([0-9]{1,})(|/)$]", $_SERVER['REQUEST_URI'])) {
-    preg_match("/[0-9]{1,}/", $_SERVER['REQUEST_URI'], $matches);
-    Books::getInstance()->booksId($matches[0]);
+$user = get($_SERVER['PHP_AUTH_USER'], false);
+$pass = get($_SERVER['PHP_AUTH_PW'], false);
+if (!$user or !$pass) {
+    header('WWW-Authenticate: Basic realm');
 }
-if (preg_match("[^/api/books/count(|/)$]", $_SERVER['REQUEST_URI'])) {
-    Books::getInstance()->count();
+$login = Books::getInstance()->auth($user, $pass);
+if (!$login) {
+    header('WWW-Authenticate: Basic realm');
+    exit('incorrect login or password');
+}
+switch ($_SERVER['REQUEST_METHOD']) {
+    case "GET":
+        if (preg_match("[^/api/books(|/)$]", $_SERVER['REQUEST_URI'])) {
+            Books::getInstance()->allBooks();

```

Comment:
All functions on this level (top API level) should return data.
Here on top level returned data from functions should be prepared for response and use it in
 `function response($message, $data)`